library/nist_kw.c

```diff
@@ -523,8 +512,9 @@ int mbedtls_nist_kw_unwrap( mbedtls_nist_kw_context *ctx,
         *out_len = 0;
     }
 
+    bad_padding = diff = 0;
```

Comment:
Note that compilers are likely to optimize these assignments away since `diff` and `bad_padding` aren't used after this point. I wouldn't even be surprised to see a compiler warn about an assignment with no effect. To avoid leaking data on the stack, use `mbedtls_platform_zeroize`.